### PR TITLE
Fix VTs hash check and add --dump-vt-verification

### PR DESCRIPTION
**Pull request by timopollmeier**
**What**:
The collation of the VTs and preferences is now set explicitly in the
SQL used to get the string the VTs verification hash is calculated from
to ensure the VTs and preferences are sorted consistently.
The unhashed string is now also generated in a function and the command
line option --dump-vt-verification has been added.

**Why**:
With the recent NVTs feed this caused the VTs verification to fail. See #1608.

**How did you test it**:
By running `gvmd --dump-vt-verification`, redirecting the output to a file and comparing it to the output of a modified version of ospd-openvas that also dumps the non-hashed verification string to a file.
I also ran gvmd --rebuild to check that warning about the mismatched hash disappeared.

**Checklist**:

- [ ] Tests
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/gvmd.c [hunk 1]
```diff
@@ -2568,6 +2573,25 @@ gvmd (int argc, char** argv)
...
+      proctitle_set ("gvmd: --dump-vt-verification");
+  
+      if (option_lock (&lockfile_checking))
+        return EXIT_FAILURE;
+
+      ret = manage_dump_vt_verification (log_config, &database);
+      log_config_free ();
+      if (ret)
+        {
+          printf ("Failed to rebuild SCAP data.\n");
```
Comment (cfi-gb) [inline src/gvmd.c:2590]: This message seems to not match what the code is actually doing.

**Review by cfi-gb — CHANGES_REQUESTED** [review]

File: src/gvmd.c (hunk 1 above)
Comment (timopollmeier) [reply src/gvmd.c]: I've fixed this in ca30e385696bb33255883bad8d6c319d5b98d8dc

**Review by bjoernricks — APPROVED** [review]

**Review by bjoernricks — APPROVED** [review]

**Comment by timopollmeier** [comment]
@mergifyio backport 21.04

**Comment by timopollmeier** [comment]
@mergifyio backport gvmd-21.04

**Pull request merged**